contracts/src/LightClient.sol

```diff
@@ -84,12 +87,21 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
     /// @notice a flag that indicates when a permissioned provrer is needed
     bool public permissionedProverEnabled;
 
-    /// @notice an array to store the L1 Block Heights where the finalizedState was updated
-    uint256[] public stateUpdateBlockNumbers;
+    ///@notice Max number of blockStates to record
+    uint64 public maxStateHistoryAllowed;
+
+    ///@notice number of block states recorded
+    uint64 public stateHistoryCount;
```

Comment:
I'm not sure that we need this variable. I think we can check if the item sitting at the first non-zero index has an L1 block number that needs to be pruned (over 10 days worth of blocks old) and if yes, remove it.

Even if we don't do it like this I think we could take the difference of the length of stateHistoryCommitments and stateHistoryFirstIndex.

Reply:
that's a good improvement - implemented here https://github.com/EspressoSystems/espresso-sequencer/commit/a2e165fc0ed456b36c0a1fe227d86baf61827cbd